dds/DCPS/transport/shmem/ShmemSendStrategy.cpp

```diff
@@ -51,10 +51,14 @@ ShmemSendStrategy::start_i()
   ShmemData* data = reinterpret_cast<ShmemData*>(mem);
   const size_t limit = (extra >= sizeof(int)) ? n_elems : (n_elems - 1);
   data[limit].status_ = ShmemData::EndOfAlloc;
-  alloc->bind(bound_name_.c_str(), mem);
+  shmem_allocator->bind(bound_name_.c_str(), mem);
+
+  {
+    ShmemDataLink::PeerAllocatorProxy proxy(*link_);
+    ShmemAllocator* shmem_allocator = proxy.peer_allocator();
+    shmem_allocator->find("Semaphore", mem);
+  }
 
-  ShmemAllocator* peer = link_->peer_allocator();
-  peer->find("Semaphore", mem);
```

Comment:
The new code calls `find` while holding a lock but the current code doesn't. Is this something to change in the master branch as well?

Reply:
Possibly.